tests/GraphBLAS-sharp.Tests/BackendCommonTests/SpMVTests.fs

```diff
@@ -0,0 +1,137 @@
+                failwith "Resulting zeroes should be implicit."
+        | None -> ()
+
+    for i in 0 .. actual.Size - 1 do
+        match actual.[i] with
+        | Some v ->
+            Expect.isTrue (isEqual v expected.[i]) $"Values should be the same. Actual is {v}, expected {expected.[i]}."
+        | None ->
+            Expect.isTrue
+                (isEqual zero expected.[i])
+                $"Values should be the same. Actual is {zero}, expected {expected.[i]}."
+
+let correctnessGenericTest
+    zero
+    sumOp
+    mulOp
+    (spMV: MailboxProcessor<_> -> Backend.CSRMatrix<'a> -> ClArray<'b option> -> ClArray<'c option>)
+    (isEqual: 'a -> 'a -> bool)
+    q
+    (testContext: TestContext)
+    (matrix: 'a [,], vector: 'a [], mask: bool [])
+    =
+
+    let mtx =
+        createMatrixFromArray2D CSR matrix (isEqual zero)
+
+    let vtr =
+        createVectorFromArray Dense vector (isEqual zero)
+
+    if mtx.NNZCount > 0 && vtr.Size > 0 then
+        try
+            let m = mtx.ToBackend testContext.ClContext
+
+            match vtr, m with
+            | VectorDense vtr, Backend.MatrixCSR m ->
+                let v = vtr.ToDevice testContext.ClContext
+
+                let res = spMV testContext.Queue m v
+
+                (Backend.MatrixCSR m).Dispose q
+                v.Dispose q
+                let hostRes = res.ToHost q
+                res.Dispose q
+
+                checkResult isEqual sumOp mulOp zero matrix vector hostRes
+            | _ -> failwith "Impossible"
+        with
+        | ex when ex.Message = "InvalidBufferSize" -> ()
+        | ex -> raise ex
+
+let testFixturesSpMV (testContext: TestContext) =
+    [ let config = defaultConfig
+      let wgSize = 32
+
+      let getCorrectnessTestName datatype = sprintf "Correctness on %s" datatype
+
+      let context = testContext.ClContext
+      let q = testContext.Queue
+      q.Error.Add(fun e -> failwithf "%A" e)
+
+      let boolSpMV =
+          Vector.spMV context boolSum boolMul wgSize
+
+      testContext
+      |> correctnessGenericTest false (||) (&&) boolSpMV (=) q
+      |> testPropertyWithConfig config (getCorrectnessTestName "bool")
+
+      let intSpMV = Vector.spMV context intSum intMul wgSize
+
+      testContext
+      |> correctnessGenericTest 0 (+) (*) intSpMV (=) q
+      |> testPropertyWithConfig config (getCorrectnessTestName "int")
+
+      let floatSpMV =
+          Vector.spMV context floatSum floatMul wgSize
+
+      testContext
+      |> correctnessGenericTest 0.0 (+) (*) floatSpMV (fun x y -> abs (x - y) < Accuracy.medium.absolute) q
+      |> testPropertyWithConfig config (getCorrectnessTestName "float")
+
+      let byteAdd =
+          Vector.spMV context byteSum byteMul wgSize
+
+      let byteToCOO = Matrix.toCOO context wgSize
+
+      testContext
+      |> correctnessGenericTest 0uy (+) (*) byteAdd (=) q
+      |> testPropertyWithConfig config (getCorrectnessTestName "byte") ]
+
+let tests =
```

Comment:
Looks like code duplication (look [here](https://github.com/YaccConstructor/GraphBLAS-sharp/pull/48/files/b250012abd937fa40d2e6bc015db74de1461d08c#diff-e13d10256497bf4a38f221c18ec706e53ebafdedde221ef060c4051b8fae87bcR73) or to other tests)